src/psyclone/domain/common/extract_driver_creator.py

```diff
@@ -0,0 +1,711 @@
+        except KeyError as err:
+            raise six.raise_from(
+                InternalError("Type '{0}' of the property reference "
+                              "'{1}' as defined in the config file "
+                              "'{2}' is not supported in the GOcean API."
+                              .format(gocean_property.intrinsic_type,
+                                      fortran_expression,
+                                      Config.get().filename)),
+                err)
+        # Handle name clashes (e.g. if the user used a variable that is
+        # the same as a flattened grid property)
+        flattened_name = symbol_table.next_available_name(flattened_name)
+        if gocean_property.type == "scalar":
+            new_symbol = DataSymbol(flattened_name, base_type)
+
+        elif gocean_property.type == "array":
+            # At this stage all gocean arrays are 2d (even integer ones)
+            # so no need to add any further tests here.
+            array = ArrayType(base_type, [ArrayType.Extent.DEFERRED,
+                                          ArrayType.Extent.DEFERRED])
+            new_symbol = DataSymbol(flattened_name, array)
+        else:
+            raise InternalError("The expression '{0}' maps to an unknown "
+                                "GOcean property type '{1}' in the config "
+                                "file '{2}'."
+                                .format(fortran_expression,
+                                        gocean_property.type,
+                                        Config.get().filename))
+
+        return new_symbol
+
+    # -------------------------------------------------------------------------
+    @staticmethod
+    def flatten_string(fortran_string):
+        '''Replaces all `%` with `_` in the string, creating a 'flattened'
+        name.
+
+        :param str fortran_string: the Fortran string containing '%'.
+
+        :returns: a flattened string (all '%' replaced with '_'.)
+        :rtype: str
+
+        '''
+        return fortran_string.replace("%", "_")
+
+    # -------------------------------------------------------------------------
+    def flatten_reference(self, old_reference, symbol_table,
+                          writer=FortranWriter()):
+        '''Replaces `old_reference` which is a structure type with a new
+        simple Reference and a flattened name (replacing all % with _).
+
+        :param old_reference: a reference to a structure member.
+        :type old_reference: \
+            :py:class:`psyclone.psyir.nodes.StructureReference`
+        :param symbol_table: the symbol table to which to add the newly \
+            defined flattened symbol.
+        :type symbol_table: :py:class:`psyclone.psyir.symbols.SymbolTable`
+        :param writer: a Fortran writer used when flattening a \
+            `StructureReference`.
+        :type writer: :py:`psyclone.psyir.backend.fortran.FortranWriter`
```

Comment:
s/:py:/:py:class:/

Reply:
Done (and there was a second one somewhere)